packages/zoe/src/contractFacet/allocationMath.js

```diff
@@ -46,7 +53,16 @@ const add = (amount, amountToAdd, _keyword) => {
 
 /** @type {Operation} */
 const subtract = (amount, amountToSubtract, keyword) => {
-  if (amountToSubtract !== undefined) {
+  // if amountToSubtract is undefined OR is defined, but empty
+  if (
+    amountToSubtract === undefined ||
+    (amountToSubtract !== undefined && AmountMath.isEmpty(amountToSubtract))
```

Comment:
Is this just equivalent to:

```suggestion
    amountToSubtract === undefined || AmountMath.isEmpty(amountToSubtract)
```

Reply:
Yes, it is. I'll simplify :)